linkerd/app/outbound/src/endpoint.rs

```diff
@@ -71,49 +67,152 @@ pub struct TcpEndpoint {
     pub labels: Option<String>,
 }
 
-impl From<listen::Addrs> for TcpLogical {
+// === impl TcpAccept ===
+
+impl From<listen::Addrs> for TcpAccept {
     fn from(addrs: listen::Addrs) -> Self {
         Self {
             addr: addrs.target_addr(),
         }
     }
 }
 
-/// Used as a default destination when resolution is rejected.
+impl Into<Addr> for &'_ TcpAccept {
+    fn into(self) -> Addr {
+        self.addr.into()
+    }
+}
+
+// === impl TcpLogical ===
+
+impl From<(Option<profiles::Receiver>, TcpAccept)> for TcpLogical {
+    fn from((profile, TcpAccept { addr }): (Option<profiles::Receiver>, TcpAccept)) -> Self {
+        Self { addr, profile }
+    }
+}
+
+/// Used for default traffic split
+impl Into<Addr> for &'_ TcpLogical {
+    fn into(self) -> Addr {
+        self.addr.into()
+    }
+}
+
+/// Used for traffic split
+impl Into<Option<profiles::Receiver>> for &'_ TcpLogical {
+    fn into(self) -> Option<profiles::Receiver> {
+        self.profile.clone()
+    }
+}
+
+/// Used to determine whether detection should be skipped.
 impl Into<SocketAddr> for &'_ TcpLogical {
     fn into(self) -> SocketAddr {
         self.addr
     }
 }
 
-/// Used to resolve endpoints.
-impl Into<Option<Addr>> for &'_ TcpLogical {
-    fn into(self) -> Option<Addr> {
-        Some(self.addr.into())
+impl std::fmt::Debug for TcpLogical {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TcpLogical")
+            .field("addr", &self.addr)
+            .field(
+                "profile",
+                if self.profile.is_some() {
+                    &"Some(..)"
+                } else {
+                    &"None"
+                },
```

Comment:
_very_ minor and probably unimportant nit: since the second argument to `DebugStruct::field` is a `&dyn fmt::Debug`, these strings will be formatted with their `Debug` impl, which includes quotes. we might want to wrap them in `format_args!` or something to format them without quotes
```suggestion
                &format_args!("{}", if self.profile.is_some() {
                    "Some(..)"
                } else {
                    "None"
                })
```